### PR TITLE
Fix #239: Explain how command parsing works

### LINKED ISSUE #239: [PE-D] wrong error message (iadd)
Should have given invalid input format error instead

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: BenedictBCJJ/ped#14

### PR DESCRIPTION
**Pull request by DrWala**
### What this does:
Explains how command parsing works to prevent confusion on how strings containing parameter-like prefixes are dealt with as well as duplicate parameter input. Thereby this fixes #239 and a whole string of similar error messages in the PE-D.

### How to test
1. cd to `docs/` folder of the project.
2. Run `bundle exec jekyll serve`.
3. Open the jekyll-hosted website in your browser (default address: http://127.0.0.1:4000).
4. Check the Command Format section's giant blue box that should explain how parameter parsing works. See if it is clear.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -40,6 +40,48 @@ Commands in this user guide follow this format:
...
+</div>
+
+<div markdown="block" class="alert alert-info">
+Commands in SunRez have a pre-set parameter list they accept. The value for the parameter will be all characters 
+until the next occurrence of a parameter prefix for the command.
+For example, if you key in this command:
+
+`radd n/John Doe p/91234567 e/e0123456@u.nus.edu y/3`
+
+The resident's name will be all characters that follows `n/` until just before the start of `p/`.
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:63]: Does this include the space?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -40,6 +40,48 @@ Commands in this user guide follow this format:
...
+**:information_source: Command parsing**<br>
+<div markdown="block" class="alert alert-info">
+All commands in SunRez default to taking the latest occurrence of a duplicate parameter **except [tags](#tag)**. 
+For example, if you key in this command:
+
+`radd n/John Doe n/Timmy Tan p/91234567 e/e0123456@u.nus.edu y/3`
+
+The resident's name will take the **later** occurrence of name and create a resident named "Timmy Tan".
+
+On the other hand, [Tags](#tag) will collate ALL tags specified in a command.
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:53]: Hmm, what do you mean by "Tags will collate ALL tags..."?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -40,6 +40,48 @@ Commands in this user guide follow this format:
...
+
+Let us break this down further and assume `radd` only takes in 2 parameters for the purposes of explanation. 
+We can break the command down as follows:
+
+`radd n/[NAME_STRING] p/[PHONE_NUM_STRING]`
+
+A user can enter a `NAME_STRING` that consists of anything, including text that contains prefix-like strings such as `s/`. 
+For example, a user could enter `John s/o Tom`. 
+However, as `s/` is not a valid prefix for the `radd` command, the command parser will treat `John s/o Tom` as the value for the name parameter. 
+The validation for the `Name` parameter will process `John s/o Tom` and reject it based on the stated validation rules.
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:73]: Should this say "... **may** reject it based on..."?

**Review by benedictkhoomw — CHANGES_REQUESTED** [review]
Overall looks good, thank you for composing an explanation of the parsing process for potentially confused users. Perhaps this should have its own sub-section entitled "Command Parsing"? It seems a little long for a note.

File: docs/UserGuide.md (hunk 2 above)
Comment (DrWala) [reply docs/UserGuide.md:53]: I wanted to specify that` t/one t/two t/three` will create a list of tags as `{one, two, three}` as opposed to taking the latest occurrence. Do you have an idea on how to improve the phrasing?

File: docs/UserGuide.md (hunk 1 above)
Comment (DrWala) [reply docs/UserGuide.md:63]: Yes. I've specified that and added that most command trim leading and trailing spaces.

File: docs/UserGuide.md (hunk 3 above)
Comment (DrWala) [reply docs/UserGuide.md:73]: Updated

**Comment by DrWala** [comment]
> Overall looks good, thank you for composing an explanation of the parsing process for potentially confused users. Perhaps this should have its own sub-section entitled "Command Parsing"? It seems a little long for a note.

I'll move it into a new section.

File: docs/UserGuide.md (hunk 2 above)
Comment (benedictkhoomw) [reply docs/UserGuide.md:54]: Maybe something like "On the other hand, all tag parameters specified will be taken. For example, ..."

Comment (DrWala) [reply docs/UserGuide.md]: Updated.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -51,7 +51,9 @@ For example, if you key in this command:
 
 The resident's name will take the **later** occurrence of name and create a resident named "Timmy Tan".
 
-On the other hand, [Tags](#tag) will collate ALL tags specified in a command.
+On the other hand, **all** [tags](#tag) specified will be taken. 
+
+For instance, `radd n/John Doe n/Timmy Tan p/91234567 e/e0123456@u.nus.edu y/3 g/tag_one g/tag_two g/tag_three` will create a resident with tags `tag_one`, `tag_two` and `tag_three`
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:56]: Is this missing a fullstop?

Comment (DrWala) [reply docs/UserGuide.md:56]: Yes.

**Review by benedictkhoomw — APPROVED** [review]
LGTM. Thank you for writing this.

**Pull request merged**